CONTRIBUTING.md

```diff
@@ -0,0 +1,142 @@
+
+## Fixes, Improvements, and Patches
+
+Fixes, improvements, and patches all happen via [GitHub Pull Requests on this repository][pulls]. If you'd like to improve the tests, you want to make the documentation clearer, you have an alternative implementation of something that may have advantages over the way its currently done, or you have any other change, we would be happy to hear about it. For trivial changes, send a pull request. For non-trivial changes, consider [opening an issue](#issues-feature-requests-and-bug-reports) to discuss it first instead.
+
+## Development
+
+### Dependencies
+
+This buildpack relies on [heroku/libcnb.rs][libcnb] to compile buildpacks. All [libcnb.rs dependencies][libcnb-deps] will need to be setup prior to building or testing this buildpack.
+
+1. Install [rust by following instructions on their site](https://www.rust-lang.org/tools/install)
+1. Follow the setup instructions on [heroku/libcnb](https://github.com/heroku/libcnb.rs)
+1. Install the [pack cli](https://buildpacks.io/docs/for-platform-operators/how-to/integrate-ci/pack/)
+
+### Building the buildpack locally
+
+1. Run `cargo check` to download dependencies and ensure there are no compilation issues.
+1. Build the buildpack:
+
+```
+cargo libcnb package
+```
+
+### Generate an application image
+
+Once you have have built the buildpack you can use `pack build` to generate an application image.
+
+```
+pack build sample-app \
+  --buildpack packaged/x86_64-unknown-linux-musl/debug/heroku_ruby \
+  --path buildpacks/ruby/tests/fixtures/default_ruby
+```
+
+This will create an image named `sample-app` based off of the fixture at `buildpacks/ruby/tests/fixtures/default_ruby`.
+
+The deployed buildpack ships with a builder that tells the `pack` CLI what other builpacks it needs. In development you must specify them via the `--buildpack` flag before this buildpack. For example to build an app that needs nodejs can run like this:
+
+```
+pack build sample-app \
+  --buildpack heroku/nodejs-engine \
+  --buildpack heroku/procfile \
+  --buildpack packaged/x86_64-unknown-linux-musl/debug/heroku_ruby \
+  --path <path/to/application>
+```
+
+List of buildpacks this buildpack depends on:
+
+```
+--buildpack heroku/nodejs-engine
+--buildpack heroku/nodejs-yarn
+--buildpack heroku/jvm
+```
+
+### Run an application image
+
+Once an image is built you can run and inspect it. Here are some example commands.
+
+- Interactive execution
+
+```
+docker run -it --rm my-image-name bash
+```
+
+- Run a command and exit
+
+```
+docker run -it --rm my-image 'which bundle'
+```
+
+- Boot the default webserver
+
+```
+docker run -it --rm --env PORT=9292 -p 9292:9292 my-image
+```
+
+- Inspect the image:
+
+```
+pack inspect my-image
+```
+
+Depending on the development situation you may need to manually instruct docker to use the CNB entrypoint. You can do that by passing `--entrypoint='/cnb/lifecycle/launcher` to your docker command.
```

Comment:
Do we need to cover this? This probably adds some confusion and is an infrequent scenario. If you happen to use a procfile, you won't need `/cnb/lifecycle/launcher`, and I think we could fix most of our buildpacks to also not need to manually `/cnb/lifecycle/launcher` too, in the same way the procfile buildpack does.